index.js

```diff
@@ -1056,7 +1056,7 @@ class Hypercore extends EventEmitter {
     try {
       if (enc) return c.decode(enc, block)
     } catch {
-      throw DECODING_ERROR()
+      throw DECODING_ERROR(undefined, this.discoveryKey)
```

Comment:
lets include err.message from the catch as message, massively helps debugging also